Add documentation for tracking progress of Cruise Control rebalances. (#11529)
strimzi/strimzi-kafka-operator: documentation/modules/cruise-control/proc-tracking-cluster-rebalance.adoc

// Module included in the following assemblies:
//
// assembly-cruise-control-concepts.adoc

[id='proc-tracking-cluster-rebalance-{context}']

= Tracking rebalances

Information surrounding the progress of an executing partition rebalance is useful for planning future cluster operations, such as worker node maintenance, scaling, or upgrading brokers.
Knowing details like the duration and data left to transfer for an ongoing partition rebalance can help users schedule maintenance windows effectively and assess the impact of continuing versus canceling a rebalance.

> Strimzi provides the following rebalance progress information fields in the `ConfigMap` listed in the `status.progress.rebalanceProgressConfigMap` of the `KafkaRebalance` resource.

Reviewer: ```suggestion
When a rebalance is in progress, Strimzi stores progress information in a `ConfigMap`.
The `ConfigMap` is referenced by the `status.progress.rebalanceProgressConfigMap` property of the `KafkaRebalance` resource .
The `ConfigMap` includes the following fields:
```

- `estimatedTimeToCompletionInMinutes`: The estimated time it will take in minutes until the partition rebalance is complete.

> - `completedByteMovementPercentage`: The percentage of the byte movement of the partition rebalance that is completed as a rounded down integer value in the range [0-100].

Reviewer: ```suggestion
- `completedByteMovementPercentage`: The percentage of data movement completed (in bytes), as a rounded down integer from 0-100.
```

- `executorState.json`: The “non-verbose” JSON payload from the link:https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control[/kafkacruisecontrol/state?substates=executor] endpoint, providing details about the executor's current status, including partition movement progress, concurrency limits, and total data to move.

> This information is updated every reconciliation of the `KafkaRebalance` resource.

Reviewer: ```suggestion
The information is updated on each reconciliation of the `KafkaRebalance` resource.
```

.Prerequisites

> * The status of the `KafkaRebalance` custom resource is `Rebalancing` state.

Reviewer: ```suggestion
* The `KafkaRebalance` custom resource is in a `Rebalancing` state.
```

// Module included in the following assemblies:
//
// assembly-cruise-control-concepts.adoc

[id='proc-tracking-cluster-rebalance-{context}']

= Tracking rebalances

You can track the progress of a partition rebalance using status information in the `KafkaRebalance` resource.
Tracking an active partition rebalance supports planning of cluster operations, including worker node maintenance, scaling, and broker upgrades. 
Understanding key details, such as duration and the remaining data to transfer, helps with scheduling maintenance windows and assessing the impact of proceeding or canceling the rebalance.

When a rebalance is in progress, Strimzi stores progress information in a `ConfigMap`.

> The `ConfigMap` is referenced by the `status.progress.rebalanceProgressConfigMap` property of the `KafkaRebalance` resource .

Reviewer 2: ... but IIRC it has a well defined name by a specific pattern (including the `KafkaRebalance` custom resource name). Maybe we can specify it here?

Author: The implementation ended up naming the `ConfigMap` with the same name as the `KafkaRebalance` resource it is referenced by. Added a note here in latest commit.

The `ConfigMap` includes the following fields:

- `estimatedTimeToCompletionInMinutes`: The estimated time it will take in minutes until the partition rebalance is complete.

- `completedByteMovementPercentage`: The percentage of data movement completed (in bytes), as a rounded down integer from 0-100.

- `executorState.json`: The “non-verbose” JSON payload from the link:https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control[/kafkacruisecontrol/state?substates=executor] endpoint, providing details about the executor's current status, including partition movement progress, concurrency limits, and total data to move.

The information is updated on each reconciliation of the `KafkaRebalance` resource.

.Prerequisites

* The `KafkaRebalance` custom resource is in a `Rebalancing` state.

.Procedure

- Accessing `estimatedTimeToCompletionInMinutes` field.:
+
[source,shell]
----

> kubectl get configmaps <my_rebalance_configmap_name> -o json | jq '.["data"]["estimatedTimeToCompletionInMinutes"]'

Reviewer 3: @kyguy These commands require additional tools (`jq`) to be installed. Is it really needed? Would something like this work for this command as well as the next ones?

```suggestion
kubectl get configmap <my_rebalance_configmap_name> -o jsonpath="{['data']['estimatedTimeToCompletionInMinutes']}"
```

Author: It's not really needed, we could replace with `jsonpath` arg of kubectl command. Just refactored.

----

- Accessing `completedByteMovementPercentage` field.:
+
[source,shell]
----
kubectl get configmaps <my_rebalance_configmap_name> -o jsonpath="{['data']['completedByteMovementPercentage']}"
----

- Accessing `executorState.json` field.
+
[source,shell]
----
kubectl get configmaps <my_rebalance_configmap_name> -o jsonpath="{['data']['executorState.json']}"

> ----

Reviewer 2: In all the above three commands I would specify the ConfigMap name differentiating the variable part (i.e. <my_rebalance>) from the fixed one. This comment is related to the one above.
